webapp/src/app.tsx

```diff
@@ -1389,16 +1389,16 @@ export class ProjectView
         // Try a quick cloud fetch. If it doesn't complete within X second(s),
         // continue on.
         const timeoutStart = Util.nowSeconds();
-        const timeout = Util.timeout(1500/*1.5 seconds*/)
+        let timedOut = false;
         p = p.then(() => {
-            return Promise.any([
-                timeout,
+            return pxt.U.promiseTimeout(
+                1500,
                 // start a partial cloud sync
                 cloud.syncAsync([h])
-                    .then((changes) => {
+                    .then(changes => {
                         if (changes.length) {
                             const elapsed = Util.nowSeconds() - timeoutStart;
-                            if (timeout.isResolved()) {
+                            if (timedOut) {
```

Comment:
So this still runs even though it timed out? Or does it cancel the callback?

If it still runs, then this looks good (if uglier haha)!

Reply:
Actually it looks like promiseTimeout is implement in terms of "Promise.race", could you just use that instead of Promise.any ?

Reply:
promises will continue to run when created / timeout doesn't stop the callback from running on any promise inside of it, it just rejects after the timeout (standard promises aren't cancellable - the bluebird implementation for cancellable promises is just a promise wrapper around other promises that rejects when cancelled, so any callbacks contained up till that point still run).

promise.isResolved is the api that doesn't exist / that timedOut is 'replacing' here, if we want can just do `elapsed < 1.5` but that feels like an impossible to debug race condition waiting to happen.

I'll see if I can make it prettier by just directly using race instead of the timeout util though ~

Reply:
Ah got it. Thanks for looking into this! Your solution also works

Reply:
You were right, promise.race came out looking better ~